module/os/linux/zfs/zfs_vnops_os.c

```diff
@@ -3965,26 +4083,30 @@ zfs_inactive(struct inode *ip)
 	}
 
 	if (zp->z_atime_dirty && zp->z_unlinked == B_FALSE) {
+		if (vfs_ratelimit_metadata_write(zfsvfs->z_os) != 0) {
+			goto out;
```

Comment:
I am not very good in Linux VFS inode lifecycle, but I have suspicion that `zfs_inactive()` may be not a good place to rate limit atime updates.  I worry that it may penalize the host system, not the user who caused it.  Please correct me if I am wrong.

Reply:
Do you mean that it will slow down inodes reuse/recycle? I'm not familiar with Linux VFS either. I was hoping someone could take the ownership of the Linux-specific code.

Reply:
Yes.  I would either remove it, or move to the point where `z_atime_dirty = B_TRUE` is set first time.